### PR TITLE
Add option to suppress warning when opening big files

### PR DESCRIPTION
**Pull request by DerivativeOfLog7**
Add an option to suppress the >=2GB file warning. Won't show on a x86 build. This is mostly useful when leaving npp searching inside files unattended as the warning could pop up and halt all progress until acted upon
image

(the original PR had 4 👍 reacts)

### CODE REVIEW

**Comment by alankilborn** [comment]
Just a thought: Maybe instead of on the `MISC.` subpage, this could go on the `Performance` subpage.  It isn't an exact fit there, but if I were looking for it, my first thought might be that subpage...

**Comment by DerivativeOfLog7** [comment]
> Just a thought: Maybe instead of on the `MISC.` subpage, this could go on the `Performance` subpage. It isn't an exact fit there, but if I were looking for it, my first thought might be that subpage...

Made it on a new branch: https://github.com/DerivativeOfLog7/notepad-plus-plus/tree/alt_placement

image

Positioning can of course be adjusted

File: PowerEditor/src/ScintillaComponent/Buffer.cpp [hunk 1]
```diff
@@ -1483,21 +1483,23 @@ bool FileManager::loadFileData(Document doc, int64_t fileSize, const TCHAR * fil
 		}
 		else // x64
 		{
+			NppGUI& nppGui = NppParameters::getInstance().getNppGUI();
+			if (!nppGui._suppress2GBWarning) {
```
Comment (donho) [inline PowerEditor/src/ScintillaComponent/Buffer.cpp:1487]: Please respect coding convention in Notepad++ :
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#general

File: PowerEditor/src/WinControls/Preference/preference.rc [hunk 2]
```diff
@@ -23,18 +23,18 @@
 #define IDC_STATIC -1
 #endif
 
-IDD_PREFERENCE_BOX DIALOGEX 0, 0, 580, 235
+IDD_PREFERENCE_BOX DIALOGEX 0, 0, 580, 250
```
Comment (donho) [inline PowerEditor/src/WinControls/Preference/preference.rc:26]: RESTORE ALL THE VALUE TO 235.

**Comment by donho** [comment]
@alankilborn 
> Just a thought: Maybe instead of on the MISC. subpage, this could go on the Performance subpage. 

For one of rare times, I agree with you totally! :)

This option, if it's not hidden, it should be in Performance subpage.

@DerivativeOfLog7 
And it should be like:

image

**Review by donho — CHANGES_REQUESTED** [review]
1. Please consider suggestions below.
2. Please consider https://github.com/notepad-plus-plus/notepad-plus-plus/pull/14053#issuecomment-1694495210

File: PowerEditor/src/WinControls/Preference/preference.rc (hunk 2 above)
Comment (DerivativeOfLog7) [reply PowerEditor/src/WinControls/Preference/preference.rc:26]: Already done on the fork

**Comment by alankilborn** [comment]
>For one of rare times, I agree with you totally! :)

Well...really...why can't we make this happen more often??  :-)

**Comment by DerivativeOfLog7** [comment]
Moved the checkbox and fixed the code style

image

**Comment by DerivativeOfLog7** [comment]
I assume other variables added for the setting should be rearranged to be near the performance settings instead of the misc?

**Comment by DerivativeOfLog7** [comment]
Would it be appropriate to add a note stating that the setting affects search in files too?

**Comment by alankilborn** [comment]
> Would it be appropriate to add a note stating that the setting affects search in files too?

A note to that effect can be added to the user manual description for the checkbox.  BTW, my personal expectation (which is meaningless) is that developers that get changes to the code accepted also write the user manual text.  (When I do a code change, I also provide the user manual change)

**Comment by donho** [comment]
image

**Comment by donho** [comment]
> 11. Finally, please test your pull requests, at least once.

Ref: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

**Comment by DerivativeOfLog7** [comment]
> > 11. Finally, please test your pull requests, at least once.
> 
> Ref: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#guidelines-for-pull-requests

Sorry... it was working when I first added it; I just forgot to reword the settings cpp after moving it to the performance tab

**Pull request closed without merging**